tests/test_rpc.py

```diff
@@ -36,6 +36,16 @@ def test_getaddress(lianad):
     assert res["address"] != lianad.rpc.getnewaddress()["address"]
 
 
+def test_listadresses(lianad):
+    list = lianad.rpc.listaddresses(2, 5)
```

Comment:
something weird here, if i call `lianad.rpc.listaddresses(0, 0)` the test will be stuck a this point (waiting indefinitely for a lianad answer):

```
tests/test_rpc.py::test_listadresses 
--------------------------------------------------------------------------------------------------------- live log setup ----------------------------------------------------------------------------------------------------------
DEBUG    root:utils.py:271 Starting 'bitcoind -datadir=/tmp/lianad-tests-bzxdc1cw/test_listadresses_1/bitcoind -printtoconsole -server -debug=1 -debugexclude=libevent -debugexclude=tor'
DEBUG    root:utils.py:357 Waiting for ['Done loading'] in the logs
DEBUG    root:utils.py:381 Found 're.compile('Done loading')' in logs
INFO     root:bitcoind.py:87 Bitcoind started
DEBUG    root:utils.py:271 Starting '/home/pyth/rust/liana_pythcoiner/tests/test_framework/../../target/debug/lianad --conf /tmp/lianad-tests-bzxdc1cw/test_listadresses_1/lianad/config.toml'
DEBUG    root:utils.py:357 Waiting for ['Database initialized and checked', 'JSONRPC server started.'] in the logs
DEBUG    root:utils.py:381 Found 're.compile('Database initialized and checked')' in logs
DEBUG    root:utils.py:381 Found 're.compile('JSONRPC server started.')' in logs
---------------------------------------------------------------------------------------------------------- live log call ----------------------------------------------------------------------------------------------------------
DEBUG    root:utils.py:208 Calling listaddresses with params (0, 0)
```
the issue seems to be `count=0`

i also try to run a lianad daemon + send `listaddresses 0 0 ` with liana-cli or [#2](https://github.com/wizardsardine/liana-rpc/pull/2) of [liana-rpc](https://github.com/wizardsardine/liana-rpc) python bindings, everything goes well, got the right answer and no panic on lianad side

Reply:
I think that's because you trigger an underflow with `max_index - 1` (see my other comment above).